In [29]:
import csv
import datetime
f = open("guns.csv","r")
data = list(csv.reader(f))
header=[]
headers=data[:1]
print(headers)
data=data[1:]
print(data[:5])

def clm_cnt(lst):
    dct={}
    for each in lst:
        if each in dct:
            dct[each]+=1
        else:
            dct[each]=1
    return dct

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']]
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [30]:
years=[each[1] for each in data]
years_count={}
years_count=clm_cnt(years)
print(years_count)

{'2014': 33599, '2012': 33563, '2013': 33636}


In [31]:
dates=[datetime.datetime(year=int(each[1]),month=int(each[2]),day=1) for each in data]
dates_count={}
dates_count=clm_cnt(dates)


In [33]:
sex=[each[5] for each in data]
sex_counts={}
sex_counts=clm_cnt(sex)
print(sex_counts)

race=[each[7] for each in data]
race_counts={}
race_counts=clm_cnt(race)
print(race_counts)

{'M': 86349, 'F': 14449}
{'Asian/Pacific Islander': 1326, 'Black': 23296, 'Native American/Native Alaskan': 917, 'Hispanic': 9022, 'White': 66237}


Observations so far:
- Amount of deaths per year is relatively stable for 2012-2014
- Number of males killed is significantly higher (86349 vs 14449 for female)
- Majority of the victims are white.

In [34]:
g = open("census.csv", "r")
census = list(csv.reader(g))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [35]:
mapping={'Asian/Pacific Islander':int(census[1][15])+int(census[1][14]),'Black':int(census[1][12]),'Hispanic':int(census[1][11]),'Native American/Native Alaskan':int(census[1][13]),'White':int(census[1][10])}
mapping

{'Asian/Pacific Islander': 15834141,
 'Black': 40250635,
 'Hispanic': 44618105,
 'Native American/Native Alaskan': 3739506,
 'White': 197318956}

In [36]:
race_per_hundredk={}
for rc,nm in race_counts.items():
    race_per_hundredk[rc] = 100000*(nm/mapping[rc])
race_per_hundredk
    

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

In [37]:
races    = [each[7] for each in data]
intents  = [each[3] for each in data]

homicide_race_counts={}
for i,rc in enumerate(races):
    if intents[i]=='Homicide':
        if rc in homicide_race_counts:
            homicide_race_counts[rc]+=1
        else:
            homicide_race_counts[rc]=1
homicide_race_counts

{'Asian/Pacific Islander': 559,
 'Black': 19510,
 'Hispanic': 5634,
 'Native American/Native Alaskan': 326,
 'White': 9147}

In [39]:
rph_homicide={}
for rc,nm in homicide_race_counts.items():
    rph_homicide[rc] = 100000*(nm/mapping[rc])
rph_homicide

{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}